aws-s3/authorization.ml

```diff
@@ -103,6 +103,50 @@ let make_auth_header ~credentials ~scope ~signed_headers ~signature =
     signed_headers
     signature
 
+let make_presigned_url ?(scheme="https") ~credentials ~date ~region ~path ~bucket ~verb ~duration () =
+  let service = "s3" in
+  let ((y, m, d), ((h, mi, s), _)) = Ptime.to_date_time date in
+  let date = sprintf "%02d%02d%02d" y m d in
+  let time = sprintf "%02d%02d%02d" h mi s in
+  let host = sprintf "%s.s3.amazonaws.com" bucket in
+  let duration = string_of_int duration in
+  let region = Region.to_string region in
+  let headers =
```

Comment:
Fold over a singleton list??? Seems a bit overkill :-)
Could it be replaced with 
```ocaml
let headers = Headers.singleton "Host" host in
``` 
?